src/com/owncloud/android/files/services/ConnectivityActionReceiver.java

```diff
@@ -0,0 +1,95 @@
+package com.owncloud.android.files.services;
+
+import android.content.BroadcastReceiver;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.net.NetworkInfo.State;
+import android.net.wifi.WifiManager;
+import android.os.Bundle;
+import android.util.Log;
+
+import com.owncloud.android.files.InstantUploadBroadcastReceiver;
+import com.owncloud.android.lib.common.utils.Log_OC;
+
+/**
+ * Receives all connectivity action from Android OS at all times and performs
+ * required OC actions. For now that are: - Signal connectivity to
+ * {@link FileUploadService}.
+ * 
+ * Later can be added: - Signal connectivity to download service, deletion
```

Comment:
Do you prefer another word? I just wanted to add a note that this is the place where it could be added... sometime, eventually, soon, in the future, or `later` :)